### DISCUSSION TITLE
Firmware Update Issue with Linux Platform

### DISCUSSION THREAD

**Discussion by yashadmiral** [Admiral-Instruments/AdmiralSquidstatAPI#2] [Q&A]
Some Linux users may encounter an issue while performing a firmware update due to inadequate permissions when attempting to connect to Squidstat devices.

Follow the steps below to create a custom rule and add the user to the appropriate groups.

1. Press **`Ctrl+Alt+t`** to open a terminal.

2. Enter the following commands:
```
sudo usermod -a -G dialout $USER
echo 'SUBSYSTEM=="hidraw", ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="003c", MODE="0660", GROUP="dialout"' | sudo tee /etc/udev/rules.d/99-admiral-instruments-squidstat.rules > /dev/null
```
3. When prompted, enter the password for the computer to enable administrative privileges.

4. When the previous command has finished executing, enter the following command:
 ```
 sudo udevadm control --reload-rules
 ```

5. If the firmware update still fails, you may need to restart your computer.

**Comment by Jace-Admiral-Instruments** [comment]
If the steps above do not resolve your issue, an alternative approach is to add an additional rule targeting the USB subsystem.
```
echo 'SUBSYSTEM=="usb", ATTRS{idVendor}=="04d8", ATTRS{idProduct}=="003c", MODE="0660", GROUP="dialout"' | sudo tee -a /etc/udev/rules.d/99-admiral-instruments-squidstat.rules > /dev/null
```
*Please restart your computer before trying to reconnect to the devices*

It is also helpful to verify that the user is in the proper group:
```
groups $USER
```
You should see `dialout` listed among the groups.


**Finally, if all else fails**, you can install the Squidstat User Interface from the [Admiral Instruments Website](https://www.admiralinstruments.com/software) and update the firmware through it instead. The latest SUI and API will have the same firmware, so once it is updated through the SUI, the API will properly connect without needing to update.
